pkg/controller/clusters/registry.go

```diff
@@ -44,9 +44,9 @@ func NewProviderClusterReconcilerRegistryBuilder() *ProviderClusterReconcilerReg
 	}
 }
 
-// Add accumulates a pair of datacenter kind a reconciler to be included in the final registry
-func (b *ProviderClusterReconcilerRegistryBuilder) Add(datacenterKind string, reconciler ProviderClusterReconciler) *ProviderClusterReconcilerRegistryBuilder {
-	b.reconciler.add(datacenterKind, reconciler)
+// Add accumulates a pair of kind and reconciler to be included in the final registry
+func (b *ProviderClusterReconcilerRegistryBuilder) Add(kind string, reconciler ProviderClusterReconciler) *ProviderClusterReconcilerRegistryBuilder {
```

Comment:
Changed this to `kind` because it seems to vary. For Snow, we focus on the the MachineConfig kind, right? @g-gaston

Reply:
My understanding was that we use datacenterKind from the Cluster spec to determine what provider to use to keep the logic we had right now. But following this thread as well.

Reply:
Yeah, we should use the datacenterkind. As vivek was saying, that's the only thing we can use to determine the provider with the current api. Not all providers have machine configs.

Reply:
Got it. Makes sense. Will change it back